src/test/java/com/googleintern/wfm/ruleengine/RuleValidationTest.java

```diff
@@ -0,0 +1,202 @@
+              .setWorkgroupId(2048L)
+              .setRoleIds(ImmutableList.of())
+              .setSkillIds(ImmutableList.of())
+              .setRoleSkillIds(ImmutableList.of())
+              .setPoolAssignments(ImmutableSet.of())
+              .build(),
+          UserModel.builder()
+              .setUserId(5)
+              .setWorkforceId(1024L)
+              .setWorkgroupId(2050L)
+              .setRoleIds(ImmutableList.of())
+              .setSkillIds(ImmutableList.of())
+              .setRoleSkillIds(ImmutableList.of())
+              .setPoolAssignments(EXPECTED_POOL_ASSIGNMENTS_USER_4)
+              .build(),
+          UserModel.builder()
+              .setUserId(6)
+              .setWorkforceId(1024L)
+              .setWorkgroupId(2060L)
+              .setRoleIds(ImmutableList.of())
+              .setSkillIds(ImmutableList.of())
+              .setRoleSkillIds(ImmutableList.of())
+              .setPoolAssignments(ImmutableSet.of())
+              .build());
+
+  private static final ImmutableSet<UserModel> USERS_WITH_LESS_ASSIGNED_PERMISSIONS =
+      ImmutableSet.of(USERS.get(5));
+
+  private static final ImmutableSet<UserModel> USERS_WITH_MORE_ASSIGNED_PERMISSIONS =
+      ImmutableSet.of(USERS.get(4));
+
+  private static final double EXPECTED_RULES_COVERAGE = (double) 5 / 7;
+
+  private static final ImmutableSet<PoolAssignmentModel> EXPECTED_UNCOVERED_POOL_ASSIGNMENTS =
+      ImmutableSet.of(
+          PoolAssignmentModel.builder().setCasePoolId(2000555L).setPermissionSetId(1112L).build());
+
+  @Test
+  public void calculateRulesCoverageTest() {
+    RuleValidation ruleValidation = new RuleValidation(USERS);
+    RuleValidationReport ruleValidationReport = ruleValidation.validate(RULES);
+    Assert.assertEquals(
+        Double.toString(EXPECTED_RULES_COVERAGE),
+        Double.toString(ruleValidationReport.ruleCoverage()));
+  }
+
+  @Test
+  public void findUncoveredPoolAssignmentsTest() {
+    RuleValidation ruleValidation = new RuleValidation(USERS);
+    RuleValidationReport ruleValidationReport = ruleValidation.validate(RULES);
+    Assert.assertEquals(
+        EXPECTED_UNCOVERED_POOL_ASSIGNMENTS, ruleValidationReport.uncoveredPoolAssignments());
+  }
+
+  @Test
+  public void findUsersWithLessAssignedPermissionsTest() {
+    RuleValidation ruleValidation = new RuleValidation(USERS);
+    RuleValidationReport ruleValidationReport = ruleValidation.validate(RULES);
+    Assert.assertEquals(
+        USERS_WITH_LESS_ASSIGNED_PERMISSIONS,
+        ruleValidationReport.usersWithLessAssignedPermissions());
+  }
+
+  @Test
+  public void findWrongUsersWithMoreAssignedPermissionsTest() {
+    RuleValidation ruleValidation = new RuleValidation(USERS);
+    RuleValidationReport ruleValidationReport = ruleValidation.validate(RULES);
+    Assert.assertEquals(
+        USERS_WITH_MORE_ASSIGNED_PERMISSIONS,
+        ruleValidationReport.usersWithMoreAssignedPermissions());
+  }
```

Comment:
Tests should not test private method.
In your case you should write a test against the validate method. 
And test the result with different input(with or without missing data).